test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -1627,14 +1662,10 @@ static int create_tm_system(void)
 
 	requirements.max_tm_queues              = NUM_TM_QUEUES + 1;
 	requirements.num_levels                 = NUM_LEVELS;
-	requirements.tm_queue_shaper_needed     = true;
-	requirements.tm_queue_wred_needed       = true;
-	requirements.tm_queue_dual_slope_needed = true;
-	requirements.vlan_marking_needed        = false;
-	requirements.ecn_marking_needed         = true;
-	requirements.drop_prec_marking_needed   = true;
-	for (color = 0; color < ODP_NUM_PACKET_COLORS; color++)
-		requirements.marking_colors_needed[color] = true;
+	/* We are not using TM queue shaping */
+	requirements.tm_queue_shaper_needed     = false;
```

Comment:
This appears incorrect since shaper_profiles[0] is set to a TM queue in create_tm_queue(). But maybe the old code is a bit obfuscated and the shaper profile is actually ODP_TM_INVALID at that point so that the shaper is actually disabled?

But even then, the validation test could still require the queue level shaper if it is supported. Node thresholds do not seem to be used either but your code enables them anyway, which may slightly improve test coverage.

The enabling could be added in the get_supported_marking_wred_threshold() function that could also be renamed at the same time (set_reqs_based_on_capas()?).

Reply:
Ack, will set it if supported based on caps